cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -55,233 +62,240 @@ const F_RETAIN_REDUNDANT_REX: u32 = 1;
         } => {
-            let encBase = iregEnc(*regBase);
-            let encIndex = iregEnc(*regIndex);
-            // The rex byte
-            let w = if clearRexW { 0 } else { 1 };
-            let r = (encG >> 3) & 1;
-            let x = (encIndex >> 3) & 1;
-            let b = (encBase >> 3) & 1;
+            let enc_base = int_reg_enc(*reg_base);
+            let enc_index = int_reg_enc(*reg_index);
+
+            // The rex byte.
+            let w = if clear_rex_w { 0 } else { 1 };
+            let r = (enc_g >> 3) & 1;
+            let x = (enc_index >> 3) & 1;
+            let b = (enc_base >> 3) & 1;
             let rex = 0x40 | (w << 3) | (r << 2) | (x << 1) | b;
-            if rex != 0x40 || retainRedundant {
+            if rex != 0x40 || retain_redundant {
                 sink.put1(rex);
             }
-            // All other prefixes and opcodes
-            while numOpcodes > 0 {
-                numOpcodes -= 1;
-                sink.put1(((opcodes >> (numOpcodes << 3)) & 0xFF) as u8);
+
+            // All other prefixes and opcodes.
+            while num_opcodes > 0 {
+                num_opcodes -= 1;
+                sink.put1(((opcodes >> (num_opcodes << 3)) & 0xFF) as u8);
             }
-            // modrm, SIB, immediates
-            if low8willSXto32(*simm32) && encIndex != regs::ENC_RSP {
-                sink.put1(mkModRegRM(1, encG & 7, 4));
-                sink.put1(mkSIB(*shift, encIndex & 7, encBase & 7));
+
+            // modrm, SIB, immediates.
+            if low8_will_sign_extend_to_32(*simm32) && enc_index != regs::ENC_RSP {
+                sink.put1(encode_modrm(1, enc_g & 7, 4));
+                sink.put1(encode_sib(*shift, enc_index & 7, enc_base & 7));
                 sink.put1(*simm32 as u8);
-            } else if encIndex != regs::ENC_RSP {
-                sink.put1(mkModRegRM(2, encG & 7, 4));
-                sink.put1(mkSIB(*shift, encIndex & 7, encBase & 7));
+            } else if enc_index != regs::ENC_RSP {
+                sink.put1(encode_modrm(2, enc_g & 7, 4));
+                sink.put1(encode_sib(*shift, enc_index & 7, enc_base & 7));
                 sink.put4(*simm32);
             } else {
-                panic!("emit_REX_OPCODES_MODRM_SIB_IMM_encG_memE: ImmRegRegShift");
+                panic!("ImmRegRegShift");
             }
         }
     }
 }
 
-/// This is the core 'emit' function for instructions that do not reference
-/// memory.
+/// This is the core 'emit' function for instructions that do not reference memory.
 ///
-/// This is conceptually the same as
-/// emit_REX_OPCODES_MODRM_SIB_IMM_encG_memE, except it is for the case
-/// where the E operand is a register rather than memory.  Hence it is much
-/// simpler.
-fn emit_REX_OPCODES_MODRM_encG_encE(
+/// This is conceptually the same as emit_modrm_sib_enc_ge, except it is for the case where the E
+/// operand is a register rather than memory.  Hence it is much simpler.
+fn emit_modrm_enc_ge(
```

Comment:
Hmm, also here is inconsistent now.

There are two axes of differentiation in the old naming:
* E is reg vs E is mem   (G is always a reg)
* whether reg values are indicated by passing a Reg ("reg") or just its encoding number ("enc")

Can you rename so as to bring back those distinctions?

Reply:
I thought I kept this distinction with the "enc" that's before "ge", and "reg" before "ge" in the other variants, but I see that I used "rm" for the Reg/Addr variant of `emit_modrm_sib_enc_ge`. Let's discuss it offline.

Reply:
We settle on `emit_std_{enc,reg}_{mem,reg}` variants here.